# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsPIDOMWindow.h
# Commit: 46af55564501
# Full Hash: 46af555645013a126416a440b658f2969eca4b20
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 55c4afa3a46c -r 46af55564501 dom/base/nsPIDOMWindow.h
--- a/dom/base/nsPIDOMWindow.h	Thu Jun 05 14:53:36 2025 +0000
+++ b/dom/base/nsPIDOMWindow.h	Thu Jun 05 14:53:36 2025 +0000
@@ -66,6 +66,7 @@
 class ServiceWorkerDescriptor;
 class Timeout;
 class TimeoutManager;
+class WebIdentityHandler;
 class WindowContext;
 class WindowGlobalChild;
 class CustomElementRegistry;
@@ -621,6 +622,8 @@
 
   bool UsingStorageAccess();
 
+  mozilla::dom::WebIdentityHandler* GetOrCreateWebIdentityHandler();
+
   uint32_t UpdateLockCount(bool aIncrement) {
     MOZ_ASSERT_IF(!aIncrement, mLockCount > 0);
     mLockCount += aIncrement ? 1 : -1;
@@ -666,6 +669,7 @@
 
   RefPtr<mozilla::dom::Performance> mPerformance;
   mozilla::UniquePtr<mozilla::dom::TimeoutManager> mTimeoutManager;
+  RefPtr<mozilla::dom::WebIdentityHandler> mWebIdentityHandler;
 
   RefPtr<mozilla::dom::Navigation> mNavigation;
 